source-git-sync-status/README.md

```diff
@@ -0,0 +1,124 @@
+  source-git, and need to be synced back to it.
+- Both of the above (source-git and dist-git history has diverged).
+
+From an operational point of view, dist-git changes will need to be synced
+back to source-git, before accepting content to be transformed from source-git
+to dist-git, in order to be able to preserve dist-git as the "blessed" place
+to store the sources. In other words: all new changes to be synced from
+source-git to dist-git should be based on a commit matching the current state
+in dist-git.
+
+In order to help with this, both update commands
+(`packit source-git update-dist-git` and `packit source-git update-source-git`)
+should refuse to update the destination repository if the history of the two
+repositories has diverged. Nevertheless, it should be possible to override
+this with a `--force` flag, to allow developers to override tooling or human
+errors and bring the repositories back in sync.
+
+## CLI proposals
+
+The naïve way to check if the two repositories are in sync is to run
+`packit source-git update-dist-git`. But this goes over the entire process to
+transform content from source-git to dist-git, which might not be practical
+when looking just for the information whether the repos are in sync, and if
+they are not, to tell the (range of) commits to be synced.
+
+Note, that the list of commits (or rather the range of commits) to be synced is
+relevant only when syncing from dist-git to source-git. Transformation in the
+other direction always considers the commits since `upstream_ref`, so in this
+case the range of outstanding commits would only serve as an information for the
+user.
+
+The CLI for the sync-check can be implemented as a new sub-command for
+`packit source-git`:
+
+```
+    $ packit source-git sync-status src/acl rpms/acl
+```
+
+The command would have the following outputs:
+
+- Print the range of commits (as a Git-revision, i.e: `HEAD~<n>..`) in
+  source-git which need to be synced to dist-git, if any.
+- Print the range of commits (as a Git-revision, i.e: `HEAD~<n>..`) in
+  dist-git which need to be synced to source-git, if any.
+- Print a message that the repositories are in sync if that's the case.
+
+## Marking the origin of synced content
+
+In order to be able to tell whether a commit in the dist-git or source-git
+history has originated from the corresponding sibling repository the _full
+hash_ of the source commit should be recorded. This can be done with some
+dedicated [Git-trailers], appended at the end of the commit messages.
+
+Commits created by `packit source-git update-dist-git` in dist-git repositories
+should have:
+
+```
+    From-source-git-commit: <head-commit-hash>
+```
+
+Commits created by `packit source-git update-source-git` in source-git
+repositories, and the commit at the tip of the branch, created by
+`packit source-git init` should have:
+
+```
+    From-dist-git-commit: <head-commit-hash>
+```
```

Comment:
+1

this will make things so much easier and more explicit

systemd is tagging source-git commits with `$VERSION-$RELEASE` part of the NVR: I like how commit IDs are indisputable